src/pool/pool.service.ts

```diff
@@ -0,0 +1,179 @@
+  async handleCron() {
+    this.logger.verbose('handleCron for pool service');
+
+    try {
+      const swaps = await this.requestDataForDay();
+      swaps.map((array) =>
+        this.logger.log(`Results length per hour ${array.length}`),
+      );
+      const flatSwaps = swaps.flat();
+      const newPools: { [key: string]: Pool } = flatSwaps.reduce(
+        (acc: { [key: string]: Pool }, swap: Swap) => {
+          try {
+            const [, poolId] = swap.id?.split(' ');
+            if (!poolId) return { ...acc };
+            let poolFromAcc = acc[poolId];
+
+            if (!poolFromAcc) {
+              poolFromAcc = new Pool();
+              poolFromAcc.id = poolId;
+              poolFromAcc.tokenFirst = swap.tokenIn;
+              poolFromAcc.tokenSecond = swap.tokenOut;
+              poolFromAcc.volume24hFirst = new Big(swap.tokenInAmount).toFixed(
+                0,
+              );
+              poolFromAcc.volume24hSecond = new Big(
+                swap.tokenOutAmount,
+              ).toFixed(0);
+            } else {
+              const [firstAmount, secondAmount] =
+                poolFromAcc.tokenFirst === swap.tokenIn
+                  ? [swap.tokenInAmount, swap.tokenOutAmount]
+                  : [swap.tokenOutAmount, swap.tokenInAmount];
+
+              poolFromAcc.volume24hFirst = new Big(poolFromAcc.volume24hFirst)
+                .add(firstAmount)
+                .toFixed(0);
+              poolFromAcc.volume24hSecond = new Big(poolFromAcc.volume24hSecond)
+                .add(secondAmount)
+                .toFixed(0);
+            }
+
+            acc[poolId] = poolFromAcc;
+            return acc;
+          } catch (e) {
+            this.logger.error(`Error during processing record ${e}`);
+            return acc;
+          }
+        },
+        {},
+      );
+
+      await this.poolRepo.save(Object.values(newPools));
+    } catch (e) {
+      this.logger.error(`Cron job error: ${e}`);
+    }
+  }
+
+  async getSwapHourSwaps(startDate, endDate) {
+    try {
+      const query = createQuery(startDate, endDate);
+      const requestData = await this.apolloClient.query({ query });
+      const swaps = requestData.data.swaps;
+      return swaps;
+    } catch (e) {
+      this.logger.error(`Error during requesting the data ${e}`);
+      return [];
+    }
+  }
+
+  async requestDataForDay() {
+    try {
+      const hourNow = new Date();
+      hourNow.setUTCMilliseconds(0);
+      hourNow.setUTCSeconds(0);
+      hourNow.setUTCMinutes(0);
+      const startTimestamp = hourNow.getTime() / 1000;
+      const resultsPerDay = Array.from(
+        { length: HOURS_IN_DAY },
+        (v, i) => startTimestamp - i * HOUR_IN_SECONDS,
+      );
+      const requests = [];
+
+      for (let i = 0; i < HOURS_IN_DAY - 1; i++) {
+        requests.push(
+          this.getSwapHourSwaps(resultsPerDay[i], resultsPerDay[i + 1]),
+        );
+      }
+
+      return Promise.all(requests).catch((e) => {
```

Comment:
You don't need to catch errors here, they will be handled in the caller method with better context.